ably/rest_client_test.go

```diff
@@ -388,6 +388,71 @@ func TestRest_hostfallback(t *testing.T) {
 	})
 }
 
+func TestRest_rememberHostFallback(t *testing.T) {
+	app, err := ablytest.NewSandbox(nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer app.Close()
+
+	t.Run("remember success host RSC15f", func(ts *testing.T) {
+		var retryCount int
+		var hosts []string
+		server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			hosts = append(hosts, r.Host)
+			retryCount++
+			w.WriteHeader(http.StatusInternalServerError)
+		}))
+		fallbacks := ably.DefaultFallbackHosts()
+		nopts := &ably.ClientOptions{
+			NoTLS:                   true,
+			FallbackHostsUseDefault: true,
+			AuthOptions: ably.AuthOptions{
+				UseTokenAuth: true,
+			},
+			HTTPClient: &http.Client{
+				Transport: &http.Transport{
+					Proxy: func(r *http.Request) (*url.URL, error) {
+						if strings.HasPrefix(r.URL.Path, "/channels/") {
+							if r.URL.Host == fallbacks[3] {
+								return url.Parse(fmt.Sprintf("https://%s", hosts[0]))
+							}
+							return url.Parse(server.URL)
+						}
+						return r.URL, nil
+					},
+				},
+			},
+		}
+		client, err := ably.NewRestClient(app.Options(nopts))
+		if err != nil {
+			ts.Fatal(err)
+		}
+		channel := client.Channels.Get("remember_fallback_host", nil)
+		err = channel.Publish("ping", "pong")
+		if err != nil {
+			ts.Fatal(err)
+		}
+		cachedHost := client.GetCachedFallbackHost()
+		if cachedHost != fallbacks[3] {
+			ts.Errorf("expected cached host to be %s got %s", fallbacks[3], cachedHost)
+		}
+		retryCount = 0
+
+		// the same cached host is used again
+		err = channel.Publish("pong", "ping")
+		if err != nil {
+			ts.Fatal(err)
+		}
+		cachedHost = client.GetCachedFallbackHost()
+		if cachedHost != fallbacks[3] {
+			ts.Errorf("expected cached host to be %s got %s", fallbacks[3], cachedHost)
+		}
+		if retryCount != 1 {
```

Comment:
I don't understand this. It looks like this test is expecting the behaviour to be:

- first call to `publish()`
  - primary host called; returns 500
  - fallback[0] called; returns 500
  - ...
  - fallback[2] called; returns 500
  - fallback[3] called; returns success
- `retry_count = 0`
- second call to `publish()`
  - .... (something happens) *
  - fallback[3] called; returns success
- check `retry_count = 1`

So that means that some failing endpoint must be called at the (something happens) point. But why? The requirement is that it would call fallback[3] without calling anything else.

Reply:
No, that's wrong. The spec says "Future HTTP requests during that period should use that host". The point of the requirement is that if you have to use a fallback, and that succeeds, then you continue to use the fallback, for all requests, for a period of x minutes before retrying the primary endpoint; the primary is only retried after that time.

Reply:
Mmh!  Okay, my bad I misunderstood what the spec meant. I do think `RSC15e` contradicts `RSC15f` though.

Reply:
> I do think `RSC15e` contradicts `RSC15f` though.

I agree - I'll add a clarification

Reply:
See https://github.com/ably/docs/commit/d3c7ce0fecea9d73c1f8c8138c22cafae743edb7